src/features/ballot/components/SubmitBallotButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useCallback } from "react";
+import { useRouter } from "next/router";
+
+import { useProjectIdMapping } from "~/features/projects/hooks/useProjects";
+import { useMaci } from "~/contexts/Maci";
+import { useBallot } from "~/contexts/Ballot";
+import { Button } from "~/components/ui/Button";
+import { Dialog } from "~/components/ui/Dialog";
+
+export const SubmitBallotButton = () => {
+  const router = useRouter();
+  const [isOpen, setOpen] = useState(false);
+  const { onVote, isLoading } = useMaci();
+  const { ballot, publishBallot } = useBallot();
+  const projectIndices = useProjectIdMapping(ballot);
+
+  const handleSubmitBallot = useCallback(async () => {
+    const votes =
+      ballot?.votes.map(({ amount, projectId }) => ({
+        voteOptionIndex: BigInt(projectIndices[projectId]),
+        newVoteWeight: BigInt(amount),
+      })) ?? [];
+
+    await onVote(
+      votes,
+      async () => {},
```

Comment:
It should have some logic for error handling or be undefined to have default handler